dlc-manager/src/contract/ord_descriptor.rs

```diff
@@ -0,0 +1,375 @@
+
+use super::{enum_descriptor::EnumDescriptor, numerical_descriptor::NumericalDescriptor};
+
+/// Descriptor for a contract involving an ordinal sat being traded.
+#[derive(Clone, Debug)]
+#[cfg_attr(
+    feature = "serde",
+    derive(Serialize, Deserialize),
+    serde(rename_all = "camelCase")
+)]
+pub struct OrdDescriptor {
+    /// The descriptor for the contract outcomes.
+    pub outcome_descriptor: OrdOutcomeDescriptor,
+    /// The original location of the ordinal within the UTXO set.
+    pub ordinal_sat_point: SatPoint,
+    /// The transaction where the UTXO containing the ordinal is located.
+    pub ordinal_tx: Transaction,
+    /// Whether it the offer party that should be refunded in case of a contract timeout.
+    pub refund_offer: bool,
+}
+
+impl OrdDescriptor {
+    pub(crate) fn get_ord_payouts(&self, total_collateral: u64) -> Result<Vec<OrdPayout>, Error> {
+        match &self.outcome_descriptor {
+            OrdOutcomeDescriptor::Enum(e) => {
+                let payouts = e.descriptor.get_payouts();
+                Ok(payouts
+                    .iter()
+                    .zip(&e.to_offer_payouts)
+                    .map(|(x, y)| OrdPayout {
+                        to_offer: *y,
+                        offer: x.offer,
+                        accept: x.accept,
+                    })
+                    .collect())
+            }
+            OrdOutcomeDescriptor::Numerical(n) => {
+                let ord_range_payouts = n.get_ord_range_payouts(total_collateral)?;
+                Ok(ord_range_payouts
+                    .iter()
+                    .map(|x| OrdPayout {
+                        to_offer: x.to_offer,
+                        offer: x.payout.payout.offer,
+                        accept: x.payout.payout.accept,
+                    })
+                    .collect())
+            }
+        }
+    }
+
+    pub(crate) fn get_postage(&self) -> u64 {
+        self.ordinal_tx.output[self.ordinal_sat_point.outpoint.vout as usize].value
+    }
+}
+
+#[derive(Clone, Debug)]
+#[cfg_attr(
+    feature = "serde",
+    derive(Serialize, Deserialize),
+    serde(rename_all = "camelCase")
+)]
+/// Descriptor for the outcomes of a contract involving an ordinal.
+pub enum OrdOutcomeDescriptor {
+    /// Descriptor for an event that has enumerated outcomes.
+    Enum(OrdEnumDescriptor),
+    /// Descriptor for an event that has numerical outcomes.
+    Numerical(OrdNumericalDescriptor),
+}
+
+#[derive(Clone, Debug)]
+#[cfg_attr(
+    feature = "serde",
+    derive(Serialize, Deserialize),
+    serde(rename_all = "camelCase")
+)]
+/// Descriptor for an event that has enumerated outcomes for contracts that involve an ordinal.
+pub struct OrdEnumDescriptor {
+    /// The description of the enumerated outcomes.
+    pub descriptor: EnumDescriptor,
+    /// The outcomes for which the ordinal is given to the offer party (one boolean for each
+    /// possible outcome).
+    pub to_offer_payouts: Vec<bool>,
```

Comment:
Could we call this "to_offeror_payouts" ? to_offer makes it sound like it's mapping something to an Offer type or something, rather than the conditions when the ordinal goes to the offeror.

Reply:
`offer` is what is used consistently in the code base and the DLC specifications so I'd rather not have it different just here.

Reply:
ok 👍